packages/build-tools/src/android/gradle.ts

```diff
@@ -35,4 +37,54 @@ export async function runGradleCommand(
     },
     env: ctx.env,
   });
+  if (ctx.isCloudBuild && process.platform === 'linux') {
+    adjustOOMScore(spawnPromise, ctx.logger);
+  }
+
+  await spawnPromise;
+}
+
+/**
+ * OOM Killer sometimes kills worker server while build is exceeding memory limits.
+ * `oom_score_adj` is a value between -1000 and 1000 that defines which process
+ * is more likely to get killed(higher value more likely).
+ *
+ * This function sets oom_score_adj for gradle process and all it's child processes.
+ */
+function adjustOOMScore(spawnPromise: SpawnPromise<SpawnResult>, logger: bunyan): void {
+  setTimeout(
+    async () => {
+      try {
+        assert(spawnPromise.child.pid);
+        const children: number[] = [spawnPromise.child.pid];
+        let shouldRetry = true;
+        while (shouldRetry) {
+          const result = await spawn('pgrep', ['-P', children.join(',')], {
+            stdio: 'pipe',
+          });
+          const pids = result.stdout
+            .toString()
+            .split('\n')
+            .map((i) => Number(i.trim()))
+            .filter((i) => i);
+          shouldRetry = false;
+          for (const pid of pids) {
+            if (!children.includes(pid)) {
+              shouldRetry = true;
+              children.push(pid);
+            }
+          }
+        }
+        await Promise.all(
+          children.map(async (pid: number) => {
+            await fs.writeFile(`/proc/${pid}/oom_score_adj`, '800\n');
```

Comment:
Maybe add a comment explaining why you're using `800` and what's the default value.

Reply:
Also, as always (xD), I'd prefer to not use the magic number. Especially, there is `\n` after the value.